cmd/scollector/collectors/continuous.go

```diff
@@ -0,0 +1,95 @@
+package collectors
+
+import (
+	"reflect"
+	"regexp"
+	"runtime"
+	"time"
+
+	"bosun.org/collect"
+	"bosun.org/metadata"
+	"bosun.org/opentsdb"
+)
+
+type ContinuousCollector struct {
+	F    func(collectorStatsChan chan<- *ContinuousCollectorStats)
+	name string
+	tags opentsdb.TagSet
+	init func()
+}
+
+type ContinuousCollectorStats struct {
+	duration float64
+	result   int
+}
+
+var ContinuousCollectorVars struct {
+	dpChan                chan<- *opentsdb.DataPoint
+	quit                  <-chan struct{}
+	reTwoOrMoreUnderscore *regexp.Regexp
+	nameWorkerChan        map[string]chan *ContinuousCollectorStats
+}
+
+func init() {
+	ContinuousCollectorVars.reTwoOrMoreUnderscore = regexp.MustCompile("[_]{2,}")
+	ContinuousCollectorVars.nameWorkerChan = make(map[string]chan *ContinuousCollectorStats)
+}
+
+func (c *ContinuousCollector) Init() {
+	if c.init != nil {
+		c.init()
+	}
+}
+
+func (c *ContinuousCollector) Run(dpChan chan<- *opentsdb.DataPoint, quit <-chan struct{}) {
```

Comment:
If you had say 50 queries, then one would have 50 collectors. Plus say you have one scollector instance that was monitoring 5 databases, that would be 250 collectors for one instance. Sure, one could make it work, but it did not seem the best solution just to make sure to use the interval collector.